sBTC/nakamoto/extensions/fixed_transaction_orders.md

```diff
@@ -0,0 +1,31 @@
+
+This document proposes that Stacks continue to support on-Bitcoin transactions for `stack-stx`, `transfer-stx`, and `delegate-stx`. However, the absence of Stacks forks makes it possible for non-finalized Stacks chain state to become inconsistent with the Bitcoin chain. For example, if Alice send 10 STX to Bob via a `stack-stx`, but the `stack-stx` is mined in a Bitcoin block that later gets orphaned, then Alice's and Bob's new balances are inconsistent with Bitcoin -- a node attempting to bootstrap from the Bitcoin and Stacks chains would not process the now-missing `stack-stx` transaction, and would be unable to authenticate the Stacks blocks against subsequent state snapshots (and thus be unable to finish booting).
+
+Bitcoin forks are rare events, and forks lasting longer than six blocks are extremely unlikely. This document proposes that a Bitcoin transaction which receives at least one tenure of Bitcoin confirmations (i.e. at least 10 Bitcoin confirmations) is sufficiently confirmed that it can be assumed that it will remain confirmed forever.
+
+Nevertheless, short-lived Bitcoin forks arise often. A naive stawman way of dealing with Stacks chain state inconsistency created by these short-lived forks is presented below for illustrative purposes:
+
+- Process all on-Bitcoin transactions at the end of the block, instead of the beginning (as it is done today). Then, the transactions in the block that are not on-Bitcoin are at least not causally-dependent on the on-Bitcoin transactions.
+- Do not process on-Bitcoin transactions that arise in tenure N until the state snapshot for tenure N is mined. This all but guarantees that these on-Bitcoin transactions will never be orphaned.
+- Report only Bitcoin data as of the last state snapshot via Clarity functions. For example, `get-burn-block-info?` would only report Bitcoin state up to the Bitcoin block which contained the last state snapshot.
+
+While this naive approach would work, the problem is that the second and third requirements introduce a very high transaction confirmation latency for Bitcoin-dependent transactions -- users would need to wait for over two tenures (over 20 Bitcoin blocks) before their Bitcoin-dependent transactions could be processed.
+
+Because Bitcoin forks are rare, this document proposes a form of speculative execution whereby Bitcoin-dependent transactions are processed as soon as they are available (and Bitcoin-dependent information exposed to Clarity as soon as available), but with the caveat that unfinalized transactions may be discarded if a fork arises. To minimize the disruption this would cause, stackers require that producers
```

Comment:
Unfortunately, that's directly from the source material. i'm not sure i have a quick fix for that one. good catch!
@will-at-stacks

Reply:
i would recommend the entire sentence be removed until a correction is known. @will-at-stacks can PR a fix once the correction is available

Reply:
rather than remove the line, i've commented it. 
it will still be visible in the raw markdown, but when rendered the comment will be invisible - @will-at-stacks you can take that commented line and PR a change when you have it. resolving this

Reply:
@wileyj sounds great!